app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,57 @@
+class PasswordResetsController < ApplicationController
+  before_action :get_user, :valid_user, :check_expiration, only: %i(edit update)
+
+  def new; end
+
+  def create
+    @user = User.find_by email: params[:password_reset][:email].downcase
+    if @user
+      @user.create_reset_digest
+      @user.send_password_reset_email
+      flash[:info] = t "reset_password.flash.msg"
+      redirect_to root_url
+    else
+      flash.now[:danger] = t "reset_password.flash.email_not_found"
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if params[:user][:password].empty?
+      @user.errors.add :password, t("reset_password.flash.password_not_empty")
+      render :edit
+    elsif @user.update_attributes user_params
+      log_in @user
+      @user.update_attribute :reset_digest, nil
+      flash[:success] = t "reset_password.flash.success"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit :password, :password_confirmation
+  end
+
+  def load_user
+    @user = User.find_by email: params[:email]
+  end
+
+  def valid_user
+    unless @user && @user.activated? && @user.authenticated?(:reset, params[:id])
```

Comment:
you can edit this line same below
`unless @user&.activated? && @user&.authenticated?(:reset, params[:id])`

Reference link: 
https://rubyinrails.com/2017/11/17/safe-navigation-operator-ampersand-dot-in-ruby/
https://stackoverflow.com/questions/37977721/why-is-safe-navigation-better-than-using-try-in-rails

Reply:
Yes, thank you